lib/src/models/parsed_complex_native_script.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'parsed_native_script.dart';
+
+part 'parsed_complex_native_script.freezed.dart';
+
+@freezed
+class ParsedComplexNativeScript with _$ParsedComplexNativeScript {
+  const factory ParsedComplexNativeScript.all({
+    required List<ParsedNativeScript> scripts,
+  }) = _All;
```

Comment:
You should rename this to ` = ParsedComplexNativeScript_All` because if you use a name starting with `_`, you can't use the dart 3 switch properly (because the class type would be private)